apps/meteor/server/lib/roles/createOrUpdateProtectedRole.ts

```diff
@@ -1,31 +1,39 @@
 import type { IRole, AtLeast } from '@rocket.chat/core-typings';
 import { Roles } from '@rocket.chat/models';
 
+import { notifyListenerOnRoleChanges } from '../../../app/lib/server/lib/notifyListenerOnRoleChanges';
+
+// No need to broadcast from here since this function will only iterate over default roles
 export const createOrUpdateProtectedRoleAsync = async (
 	roleId: string,
 	roleData: AtLeast<Omit<IRole, '_id' | 'protected'>, 'name'>,
 ): Promise<void> => {
 	const role = await Roles.findOneById<Pick<IRole, '_id' | 'name' | 'scope' | 'description' | 'mandatory2fa'>>(roleId, {
 		projection: { name: 1, scope: 1, description: 1, mandatory2fa: 1 },
 	});
+
 	if (role) {
-		await Roles.updateById(
+		const updatedRole = await Roles.updateById(
 			roleId,
 			roleData.name || role.name,
 			roleData.scope || role.scope,
 			roleData.description || role.description,
 			roleData.mandatory2fa || role.mandatory2fa,
 		);
 
+		void notifyListenerOnRoleChanges(updatedRole._id);
```

Comment:
since `Roles.updateById` returns the updated value, we can use it to skip an additional round trip to DB
```suggestion
		void notifyListenerOnRoleChanges(updatedRole._id, 'updated', updatedRole);
```